tests/test_pvc_concurrent_deletion_creation.py

```diff
@@ -0,0 +1,260 @@
+        pv = verify_pvc_and_fetch_pv_name(pvc_name)
+        assert pv, f'PVC {pvc_name} does not exists'
+        self.initial_pvs.append(pv)
+
+
+def teardown(self):
+    """
+    Delete PVCs
+    Delete storage class
+    Delete temporary yaml files
+    Delete project
+    """
+    # Delete newly created PVCs
+    command = (f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
+               f'{self.pvc_base_name_new}$i -n {TEST_PROJECT};done')
+    proc = run_async(command)
+    assert proc, (f'Failed to execute command for deleting '
+                  f'{self.number_of_pvc} PVCs')
+
+    # Verify command to delete PVCs
+    ret, _, _ = proc.async_communicate()
+    assert not ret, "Deletion of newly created PVCs failed"
+
+    assert SC.delete(yaml_file=TEMP_YAML_FILE_SC)
+    assert run_cmd(f"oc delete project {TEST_PROJECT}")
+    delete_file(TEMP_YAML_FILE_SC)
+
+
+def create_storage_class(**kwargs):
+    """
+    Create storage class
+
+    Args:
+        **kwargs: key, value pairs for yaml file substitution
+    """
+    sc_yaml = os.path.join(TEMPLATE_DIR, "storageclass.yaml")
+    sc_name = kwargs['storageclass_name']
+
+    sc_yaml_file = templating.generate_yaml_from_jinja2_template_with_data(
+        sc_yaml, **kwargs
+    )
+    with open(TEMP_YAML_FILE_SC, 'w') as yaml_file:
+        yaml.dump(sc_yaml_file, yaml_file, default_flow_style=False)
+    log.info(f'Creating Storage Class {sc_name}')
+    assert SC.create(yaml_file=TEMP_YAML_FILE_SC)
+    log.info(f'Created Storage Class {sc_name}')
+
+
+def create_multiple_pvc(pvc_base_name, number_of_pvc):
+    """
+    Create PVCs
+
+    Args:
+        pvc_base_name (str): Prefix of PVC name
+        number_of_pvc (int): Number of PVCs to be created
+    """
+    for count in range(1, number_of_pvc + 1):
+        pvc_name = f'{pvc_base_name}{count}'
+        log.info(f'Creating Persistent Volume Claim {pvc_name}')
+        PVOLC.name = pvc_name
+        assert PVOLC.create_pvc(storageclass='test-sc')
+        log.info(f'Created Persistent Volume Claim {pvc_name}')
+
+
+def run_async(command):
+    """
+    Run command locally and return without waiting for completion
+
+    Args:
+        command (str): The command to run on the system.
+
+    Returns:
+        An open descriptor to be used by the calling function.
+        None on error.
+    """
+    log.info(f"Executing command: {command}")
+    p = subprocess.Popen(
+        command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, shell=True
+    )
+
+    def async_communicate():
+        stdout, stderr = p.communicate()
+        retcode = p.returncode
+        return retcode, stdout, stderr
+
+    p.async_communicate = async_communicate
+    return p
+
+
+def verify_pvc_and_fetch_pv_name(pvc_name):
+    """
+    Verify that the status of PVC is 'Bound' and fetch PV name
+
+    Args:
+        pvc_name (str): Name of PVC
+
+    Returns:
+        PV name if PV exists, None if PVC does not exists.
```

Comment:
Change to:
str: PV name if PV exists, None if PVC does not exist

Reply:
Done